test/metabase/util/memoize_test.cljc

```diff
@@ -0,0 +1,73 @@
+(ns metabase.util.memoize-test
+  (:require
+   [clojure.string :as str]
+   [clojure.test :refer [deftest is testing]]
+   [clojure.test.check.generators :as gen]
+   [metabase.util.memoize :as memo]))
+
+(defn- call-count-harness [memoizer inner-fn]
+  (let [misses        (atom 0)
+        calls         (atom 0)
+        wrapped-inner (fn [x]
+                        (swap! misses inc)
+                        (inner-fn x))
+        memoized      (memoizer wrapped-inner)
+        wrapped-memo  (fn [x]
+                        (swap! calls inc)
+                        (memoized x))]
+    {:misses misses
+     :calls  calls
+     :f      wrapped-memo
+     :reset  (fn []
+               (reset! misses 0)
+               (reset! calls  0))}))
+
+(defn- round-robin [times keyspace f exp-fn]
+  (let [exp-map (into {} (map (juxt identity exp-fn)) keyspace)]
+    (dotimes [_ times]
+      (doseq [k keyspace]
+        (is (= (get exp-map k ::not-found)
+               (f k)))))))
+
+(defn- never-evicts [memoizer]
+  (let [keyspace                 (vec (set (gen/sample gen/string 100)))
```

Comment:
Why not use `gen/set`?

Reply:
I didn't know you could pass it the size.